### ISSUE TITLE
integration flake: kubectl delete deployment/nginx-deployment

### ISSUE THREAD

**Issue by gmarek** [kubernetes/kubernetes#21257]
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/21058/kubernetes-pull-test-unit-integration/14383/build-log.txt

```
Successful get deployment nginx-deployment {{.spec.replicas}}: 1
(BI0215 02:36:58.517962   30654 event.go:211] Event(api.ObjectReference{Kind:"Deployment", Namespace:"default", Name:"nginx-deployment", UID:"09b3c8a2-d3d0-11e5-a35b-0242ac110004", APIVersion:"extensions", ResourceVersion:"1240", FieldPath:""}): type: 'Normal' reason: 'ScalingReplicaSet' Scaled down replica set nginx-deployment-r5u5h to 1
I0215 02:36:58.550730   30654 event.go:211] Event(api.ObjectReference{Kind:"ReplicaSet", Namespace:"default", Name:"nginx-deployment-r5u5h", UID:"09b4bda1-d3d0-11e5-a35b-0242ac110004", APIVersion:"extensions", ResourceVersion:"1241", FieldPath:""}): type: 'Normal' reason: 'SuccessfulDelete' Deleted pod: nginx-deployment-r5u5h-k40co
I0215 02:36:58.550842   30654 event.go:211] Event(api.ObjectReference{Kind:"ReplicaSet", Namespace:"default", Name:"nginx-deployment-r5u5h", UID:"09b4bda1-d3d0-11e5-a35b-0242ac110004", APIVersion:"extensions", ResourceVersion:"1241", FieldPath:""}): type: 'Normal' reason: 'SuccessfulDelete' Deleted pod: nginx-deployment-r5u5h-in3wz
Error from server: deployments.extensions "nginx-deployment" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
!!! Error in ./hack/test-cmd.sh:1004
  'kubectl delete deployment/nginx-deployment "${kube_flags[@]}"' exited with status 1
```

It looks that we lack retry in the test.

cc @kubernetes/goog-ux

**Comment by gmarek** [comment]
As can be seen by number of references, this seems to be happening often. Bumping to P0.

**Comment by wojtek-t** [comment]
@nikhiljindal

**Comment by janetkuo** [comment]
Fix #21346 is waiting to be merged.
